File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/users/UsersGetPresenceRequest.java
package com.github.seratch.jslack.api.methods.request.users;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

public class UsersGetPresenceRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `users:read`
     */
    private String token;

    /**
     * User to get presence info on. Defaults to the authed user.
     */
    private String user;

    UsersGetPresenceRequest(String token, String user) {
        this.token = token;
        this.user = user;
    }

    public static UsersGetPresenceRequestBuilder builder() {
        return new UsersGetPresenceRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public String getUser() {
        return this.user;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public static class UsersGetPresenceRequestBuilder {
        private String token;
        private String user;

        UsersGetPresenceRequestBuilder() {
        }

        public UsersGetPresenceRequest.UsersGetPresenceRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public UsersGetPresenceRequest.UsersGetPresenceRequestBuilder user(String user) {
            this.user = user;
            return this;
        }

        public UsersGetPresenceRequest build() {
            return new UsersGetPresenceRequest(token, user);
        }

        public String toString() {
            return "UsersGetPresenceRequest.UsersGetPresenceRequestBuilder(token=" + this.token + ", user=" + this.user + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/users/UsersGetPresenceRequest.kt
package work.socialhub.kslack.api.methods.request.users

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

@JsExport
class UsersGetPresenceRequest(
    /** Authentication token. Requires scope: `users:read` */
    override var token: String?,
    /** User to get presence info on. Defaults to the authed user. */
    var user: String?
) : SlackApiRequest, FormRequest{

    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {
            it.addParam("user", user)
        }
    }
}